### ISSUE TITLE
2.2.1 seems to have introduced an issue that breaks output of istanbul

### ISSUE THREAD

**Issue by danmartinez101** [jasmine/jasmine-npm#36]
When running with version 2.2.0 istanbul coverage is displayed at the end of a jasmine run. Upgrading to 2.2.1 is apparently killing istanbul before it has completed. This results in no console output, but more importantly, when running on jenkins, the coverage output file are incomplete resulting in false positives(i.e. build passes even when a file doesn't have proper code coverage)

**Comment by danmartinez101** [comment]
```
Daniels-MacBook-Pro:haakon-core dmartinez$ npm install jasmine@2.2.0
jasmine@2.2.0 node_modules/jasmine
├── jasmine-core@2.2.0
└── glob@3.2.11 (inherits@2.0.1, minimatch@0.3.0)
Daniels-MacBook-Pro:haakon-core dmartinez$ ./node_modules/.bin/istanbul cover -x 'spec/support/**' ./spec/support/runner.js
Failed to load c++ bson extension, using pure JS version
Started
..................................


34 specs, 0 failures
Finished in 0.052 seconds
=============================================================================
Writing coverage object [/Users/dmartinez/source/haakon-core/coverage/coverage.json]
Writing coverage reports at [/Users/dmartinez/source/haakon-core/coverage]
=============================================================================

=============================== Coverage summary ===============================
Statements   : 100% ( 469/469 )
Branches     : 100% ( 0/0 )
Functions    : 100% ( 155/155 )
Lines        : 100% ( 469/469 )
================================================================================
Daniels-MacBook-Pro:haakon-core dmartinez$ npm uninstall jasmine
unbuild jasmine@2.2.0
Daniels-MacBook-Pro:haakon-core dmartinez$ npm install jasmine@2.2.1
jasmine@2.2.1 node_modules/jasmine
├── exit@0.1.2
├── jasmine-core@2.2.0
└── glob@3.2.11 (inherits@2.0.1, minimatch@0.3.0)
Daniels-MacBook-Pro:haakon-core dmartinez$ ./node_modules/.bin/istanbul cover -x 'spec/support/**' ./spec/support/runner.js
Failed to load c++ bson extension, using pure JS version
Started
..................................


34 specs, 0 failures
Finished in 0.048 seconds
Daniels-MacBook-Pro:haakon-core dmartinez$ 
```

**Comment by slackersoft** [comment]
Interesting, the only change between 2.2.0 and 2.2.1 is the introduction of the exit module to allow jasmine to properly wait for stdout to flush on windows when run stdout is being piped.

That said, I think this has kind of problem has been fixed on master. We're now doing a better job of making sure that the reporter that causes the process to exit is the last reporter to run at the end of the suite. If you have a chance to check it out, I'd like to know if this fixes your issue as well.

Thanks for using jasmine!

**Comment by danmartinez101** [comment]
@slackersoft I gave master a shot and it seems to still be an issue :/

```
Daniels-MacBook-Pro:haakon-core dmartinez$ npm install jasmine@2.2.0
jasmine@2.2.0 node_modules/jasmine
├── jasmine-core@2.2.0
└── glob@3.2.11 (inherits@2.0.1, minimatch@0.3.0)
Daniels-MacBook-Pro:haakon-core dmartinez$ npm run test

> haakon-core@0.0.1 test /Users/dmartinez/source/haakon-core
> istanbul cover ./spec/support/runner.js

Failed to load c++ bson extension, using pure JS version
Started
....................................


36 specs, 0 failures
Finished in 0.058 seconds
=============================================================================
Writing coverage object [/Users/dmartinez/source/haakon-core/coverage/coverage.json]
Writing coverage reports at [/Users/dmartinez/source/haakon-core/coverage]
=============================================================================

=============================== Coverage summary ===============================
Statements   : 100% ( 515/515 )
Branches     : 100% ( 0/0 )
Functions    : 100% ( 170/170 )
Lines        : 100% ( 515/515 )
================================================================================
Daniels-MacBook-Pro:haakon-core dmartinez$ npm uninstall jasmine
unbuild jasmine@2.2.0
Daniels-MacBook-Pro:haakon-core dmartinez$ npm install git+https://github.com/jasmine/jasmine-npm.git
jasmine@2.2.1 node_modules/jasmine
├── exit@0.1.2
├── jasmine-core@2.2.0
└── glob@3.2.11 (inherits@2.0.1, minimatch@0.3.0)
Daniels-MacBook-Pro:haakon-core dmartinez$ npm run test

> haakon-core@0.0.1 test /Users/dmartinez/source/haakon-core
> istanbul cover ./spec/support/runner.js

Failed to load c++ bson extension, using pure JS version
Started
....................................


36 specs, 0 failures
Finished in 0.058 seconds
Daniels-MacBook-Pro:haakon-core dmartinez$ 
```

**Comment by slackersoft** [comment]
After a bit more digging, it looks like the [exit module](https://npmjs.org/package/exit) that we started using to fix #20 also has another windows specific fix for making sure the error code is set properly that is probably causing this.

We're looking into a better solution that ensures proper error codes and works with piped output on windows. We'd also be happy to review a pull request that solves this issue.

**Comment by amavisca** [comment]
Fixed with 97c67d5a0888b2694069dc56403e170203bad365
